_std/macros/istraits_or_genetics.dm

```diff
@@ -0,0 +1,2 @@
+//for trait / genetics checks
+#define isalcoholresistant(x) ((x.bioHolder && x.bioHolder.HasEffect("resist_alcohol")) || (x.traitHolder && x.traitHolder.hasTrait("training_drinker")))
```

Comment:
```suggestion
#define isalcoholresistant(x) ((x).bioHolder?.HasEffect("resist_alcohol") || (x).traitHolder?.hasTrait("training_drinker"))
```
Let's clean up the macro while we're here.

Reply:
Also I feel like the name of this file is way too specific. But I'm not sure what else would I name such file. Someone else chime in.

Reply:
genetics_traits.dm?

Reply:
good call! tthat sounds better